lib/phidgets.js

```diff
@@ -2666,3 +2668,361 @@ PhidgetRFID.prototype.write = function(tag, protocol, lock) {
+  
+  var self = this;
+
+  // keywords which should result in an emitted event
+  var emittableEvents = [
+
+    // @TODO is this event necessary?
+    /**
+     * The sensor's ambient temperature has changed.
+     *
+     * @event ambientTemperature
+     * @param {PhidgetTemperatureSensor} emitter The actual PhidgetTemperatureSensor object that emitted the
+     * event.
+     * @param {Object} data An object containing the sensor data and related information
+     * @param {Number} data.value The sensor's new ambient temperature.
+     */
+    'AmbientTemperature',
+
+    /**
+     * The calculated temperature of the given input has changed by more than temperatureChangeTrigger.
+     * This value is dependent on the sensor's ambient temperature, the input's thermocouple type, and
+     * the input's potential.
+     *
+     * @event temperature
+     * @param {PhidgetTemperatureSensor} emitter The actual PhidgetTemperatureSensor object that emitted the
+     * event.
+     * @param {Object} data An object containing the sensor data and related information
+     * @param {int} data.index The input's index number
+     * @param {Number} data.value The input's new temperature.
+     */
+    'Temperature',
+
+    // @TODO it may be useful to return the last known AmbientTemperature with this event, as its
+    // most likely use is in calculating temperature from a voltage source other than a J K E or T thermocouple.
+    /**
+     * The potential (voltage) of the given input has changed.
+     *
+     * @event potential
+     * @param {PhidgetTemperatureSensor} emitter The actual PhidgetTemperatureSensor object that emitted the
+     * event.
+     * @param {Object} data An object containing the sensor data and related information
+     * @param {int} data.index The input's index number
+     * @param {Number} data.value The input's new potential.
+     */
+    'Potential'
+  ];
+
+  // keywords which relate to a specific input, and have a useful data.index value
+  var inputSpecificEvents = [
+    'Temperature',
+    'TemperatureMax',
+    'TemperatureMin',
+    'Potential',
+    'PotentialMax',
+    'PotentialMin',
+    'ThermocoupleType',
+    'Trigger'
+  ];
+
+  // emitted event names and properties on each input must be camelCase
+  var camelizedKeyword = data.keyword.charAt(0).toLowerCase() + data.keyword.slice(1);
+
+  // the value emitted by an emittableEvent
+  var eventData = {
+    value: Number(data.value)
+  };
+
+  if (data.keyword === 'AmbientTemperature') {
+    self._ambientTemperature = eventData.value;
+
+  } else if (data.keyword === 'AmbientTemperatureMax') {
+    self._ambientTemperatureMax = eventData.value;
+
+  } else if (data.keyword === 'AmbientTemperatureMin') {
+    self._ambientTemperatureMin = eventData.value;
+
+  } else if (data.keyword === 'NumberOfSensors') {
+    self.inputs.count = eventData.value;
+
+  } else if (inputSpecificEvents.indexOf(data.keyword) >= 0) {
```

Comment:
The property `PhidgetTemperatureSensor#inputs[index].trigger` is simply `.trigger` because it DRY'd up the `_parsePhidgetSpecificData()` handling. It could be changed, but would require less-general/more-specific logic in this function.